storage/io/src/main/java/org/openscience/cdk/io/SDFWriter.java

```diff
@@ -126,6 +132,18 @@ public SDFWriter(Set<String> propertiesToWrite) {
         this(new StringWriter(), propertiesToWrite);
     }
 
+    /**
+     * Ensures a writer is buffered.
+     *
+     * @param wtr writer, may be buffered
+     * @return a BufferedWriter
+     */
+    private static BufferedWriter ensureBuffered(Writer wtr) {
```

Comment:
What about the idea to have the class implement BufferedReader too? One complication with that would be that most of our readers support Reader and InputStream. The latter is at least import for XML-based formats...